lib/tarantool_server.py

```diff
@@ -449,7 +449,7 @@ def seek_wait(self, msg, proc=None, name=None):
         while True:
             if os.path.exists(self.path):
                 break
-            time.sleep(0.001)
+            gevent.sleep(0.001)
```

Comment:
To be honest, lack of negative test cases disquiets me more and more. We unlikely will break the general case, when all tests are passed, becase every test-run push is verified against tarantool's test suites (manually, by me). I'm not so sure about a negative and tricky cases like this one. Maybe it is good time to start collecting negative test cases somewhere in the repository? Even if it'll require some manual work to run and verify, it'll be a good start.

I would add a directory with negative test cases and would add some simple runner (Makefile / shell script / pytest — does not matter), which would do (sketchy code, I didn't run it):

```sh
#!/bin/sh

set -eu

cd negative_testing
rm -rf output
mkdir output
for test in */*.test.lua; do
    printf 'Starting %s\n' "${test}"

    suite="${test%/*}"
    output_file="output/${test%.test.lua}.out.txt"
    validate_script="${test%.test.lua}/.validate.sh"

    mkdir -p "output/${suite}"
    ./test-run.py "${test}" > "${output_file}" 2>&1 || true
    "./${validate_script}" < "${output_file}"

    printf 'The test %s passed\n' "${test}"
done
```

negative_testing/foo/bar.validate.sh would contain a script that verifies an output of test-run. Just grep in the simplest case.

Reply:
(Or at least track it with an issue.)

Reply:
Fixed your script and added pytest script for comparing gold log and test-run log. Does it seem like an instrument which you want?

Reply:
It is definitely something toward the negative testing I want. But I don't see any test case. Can you add a test case for the issue of the topic? Otherwise I'll unable to experiment around the instrument and give you a feedback.

While we are here, several preliminary points:

1. Are we really need two runners: a shell script and pytest? If we use pytest here, I would start test-run.py from pytest directly (as a subprocess).
2. I'm under heavy impression that the approach with comparison of all output against a gold result leads to many problems. I would check the output for presence of an expected hunk (certain error message or so).

Reply:
I update a new variant without reconciliation between logs.  Now we wait for expected lines, which can explain the current state test test-run log.

Reply:
Just resolving. Feel free to reopen.